test/fluree/db/transact/retraction_test.clj

```diff
@@ -41,4 +41,36 @@
                             {:context [test-utils/default-context
                                        {:ex "http://example.org/ns/"}],
                              :select {:ex/alice [:*]}}))
-          "Alice should no longer have an age property"))))
+          "Alice should no longer have an age property")))
+  (testing "retracting ordered lists"
+    (let [conn             (test-utils/create-conn)
+          ledger           @(fluree/create conn "tx/retract")
+          context          [test-utils/default-str-context
+                            {"ex"        "http://example.org/ns/"
+                             "ex:items2" {"@container" "@list"}}]
+          q1               {:context context
+                            :select  {"ex:list-test" ["*"]}}
+          db               @(fluree/stage
+                             (fluree/db ledger)
+                             {"@context" context
+                              "insert"
+                              [{"id"        "ex:list-test"
+                                "ex:items1" {"@list" ["zero" "one" "two"
+                                                      "three"]}
+                                "ex:items2" ["four" "five" "six" "seven"]}]})
+          before-retract   @(fluree/query db q1)
+          db-after-retract @(fluree/stage
+                             db
+                             {"@context" context
+                              "delete"   {"id"        "ex:list-test"
+                                          "ex:items1" "?items1"
+                                          "ex:items2" "?items2"}
+                              "where"    {"id"        "ex:list-test"
+                                          "ex:items1" "?items1"
+                                          "ex:items2" "?items2"}})
+          after-retract    @(fluree/query db-after-retract q1)]
```

Comment:
I think your `after-retract` needs to be against a db that's been committed and re-loaded in order to exercise the `merge-commit` call stack.